Move fsnotify.hh to the 'include' directory for public use.

balusch opened the pull request:
Signed-off-by: Jianyong Chen <baluschch@gmail.com>

xemul commented:
Please, put the whole thing into `namespace experimental` so that we could tweak the API before freezing it. Also, add the doxygen-doc comments to public classes/methods/helpers

balusch commented:
@xemul Hi, Pavel! I have moved it into the `experimental` namespace and added some doxygen doc based on the original comments as well as the man page of `inotify`, please have a look again. And since I'm not a native English speaker and am new to Doxygen, please feel free to correct me if there is anything wrong or can be improved.

balusch commented:
ping

nyh commented on include/seastar/core/fsnotify.hh:
```diff
@@ -80,26 +90,41 @@ public:
 
+        /// Reset the watch point.
+        ///
+        /// \note Note that this operation won't unregistered the event for
```
typo: unregister

balusch replied:
done

nyh reviewed (approved):
Looks good to me. @elcallio as the original author of this code, please review as well.

elcallio commented:
Looks good to me. Kudos on the comment/doc cleanups. But note that @avikivity insisted we kept this API internal at the time, so maybe he should weigh in as well?

balusch commented:
> Looks good to me. Kudos on the comment/doc cleanups. But note that @avikivity insisted we kept this API internal at the time, so maybe he should weigh in as well?

Sure. @avikivity please also have a look at this patch. Although I don't know why you insist on keeping this API internal in the outset, I guess we could make it public now since it is placed in the `experimental` namespace; thus we could tweak it in the future as Pavel said.

xemul commented:
@avikivity , do you have any objections against this PR?
